sys/include/net/ieee802154/radio.h

```diff
@@ -0,0 +1,1085 @@
+     *
+     * This event is present if radio has @ref IEEE802154_CAP_IRQ_CCA_DONE.
+     */
+    IEEE802154_RADIO_CONFIRM_CCA,
+} ieee802154_trx_ev_t;
+
+/**
+ * @brief CSMA-CA exponential backoff parameters.
+ */
+typedef struct {
+    uint8_t min;    /**< minimum value of the exponential backoff */
+    uint8_t max;    /**< maximum value of the exponential backoff */
+} ieee802154_csma_be_t;
+
+/**
+ * @brief RX information associated to a frame
+ */
+typedef struct {
+    int16_t rssi;   /**< RSSI of the received frame */
+    uint8_t lqi;    /**< LQI of the received frame */
+} ieee802154_rx_info_t;
+
+/**
+ * @brief Transmission flags.
+ */
+typedef enum {
+    /**
+     * @brief transmission uses Auto CCA or frame retransmissions with CSMA-CA
+     */
+    TX_MODE_AUTO=(1 << 0),
+    /**
+     * @brief the transceiver goes immediately to RX state after TX done.
+     */
+    TX_MODE_TXRX=(1 << 1),
+} ieee802154_tx_flags_t;
+
+/**
+ * @brief TX information of the last transmitted packet.
+ */
+typedef struct {
+    ieee802154_tx_flags_t tx_flags; /**< flags of the last transmission */
+    int8_t retries;                 /**< number of frame retransmissions of the last TX */
+} ieee802154_tx_info_t;
+
+/**
+ * @brief Forwared declaration of the IEEE802.15.4 device descriptor
+ */
+typedef struct ieee802154_dev ieee802154_dev_t;
+
+/**
+ * @brief Prototype of the IEEE802.15.4 device event callback
+ *
+ * @param[in] dev IEEE802.15.4 device descriptor
+ * @param[in] status the status
+ */
+typedef void (*ieee802154_cb_t)(ieee802154_dev_t *dev,
+                                ieee802154_trx_ev_t status);
+
+/**
+ * @brief the IEEE802.15.4 device descriptor
+ */
+struct ieee802154_dev {
+    /**
+     * @brief pointer to the operations of the device
+     */
+    const ieee802154_radio_ops_t *driver;
+    /**
+     * @brief pointer to the context of the device
+     */
+    void *ctx;
+    /**
+     * @brief the event callback of the device
+     */
+    ieee802154_cb_t cb;
+};
+
+/**
+ * @brief IEEE802.15.4 CCA modes
+ */
+typedef enum {
+    IEEE802154_CCA_MODE_ED_THRESHOLD,       /**< CCA using first mode (energy detection) */
+    IEEE802154_CCA_MODE_CARRIER_SENSING,    /**< CCA using second mode (carrier sensing) */
+    IEEE802154_CCA_MODE_ED_THRESH_AND_CS,   /**< CCA using third mode (energy detection AND carrier sensing) */
+    IEEE802154_CCA_MODE_ED_THRESH_OR_CS,    /**< CCA using third mode (energy detection OR carrier sensing) */
+} ieee802154_cca_mode_t;
+
+/**
+ * @brief Auto ACK configuration
+ */
+typedef enum {
+    IEEE802154_AACK_CONF_DISABLED,      /**< Auto ACK is disabled */
+    IEEE802154_AACK_CONF_ENABLED,       /**< Auto ACK is enabled */
+    IEEE802154_AACK_CONF_FRAME_PENDING, /**< Auto ACK is enabled and frame pending bit set in the next ACK packet */
+} ieee802154_aack_conf_t;
+
+/**
+ * @brief Holder of the PHY configuration
+ */
+typedef struct {
+    uint8_t channel;    /**< IEEE802.15.4 channel number */
```

Comment:
channel can be > 255 for FSK and OFDM

![image](https://user-images.githubusercontent.com/1301112/89218568-9485d700-d5ce-11ea-8981-1e72375c1e25.png)

Reply:
Hmmm maybe uint16_t then

Reply:
https://github.com/RIOT-OS/RIOT/pull/14371/commits/f5cea88e3ff99e78fe8df34ac2c850973fc6e753